README.md

```diff
@@ -347,6 +347,7 @@ config
   .context(context)
   .externals(externals)
   .loader(loader)
+  .mode(mode)
```

Comment:
The config merging section of the README also references the top level options, could you add `mode` there too? (I see `parallelism` is also missing from that, perhaps worth fixing whilst we're there)

Also, could you add the `optimization` options to the README too? :-)